### DISCUSSION TITLE
Azure resource naming not following Azure recommended guidelines?

### DISCUSSION THREAD

**Discussion by baatch** [Azure/data-management-zone#188] [General] [answered]
Hi, 

Wondering why the Azure resources deployed in the templates does not follow your own Azure naming recommendations in here: https://docs.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-abbreviations
?

**Comment by marvinbuss** [comment, marked as answer]
The naming convention proposed by CAF is an option that customers can take a dependency on, but don't have to. Most projects do not take a dependency on the naming convention today.

In general, we believe that  centralized naming conventions are not required or need to be enforced across a tenant. Users or customers should rather use and enforce resource tags to quickly and easily find the right resources that they are looking for. If a landing zone team (not DATA landing zone, but landing zone in general) decides to use their own naming convention, they should be free to do so. All resources within an environment have sufficient properties and tags to identify them quickly and securely. In summary, we don't see the naming convention as a critical design area and tags should be used and enforced tags via policy (i.e. for cost management reasons, to specify the environment, etc.) if additional context is required in any environment.

One of the reasons why we are not using the convention in the link above is that all the details would drastically reduce the allowed size of the prefix. This would make it more difficult for users to find a prefix that is not already taken and therefore may hinder some of the deployments. This would then also hinder adoption of ESA. In addition, we believe that some of the proposed sections are not required. For example, the region of services and artifacts is put front and center even the default table view in the Azure Portal.

Today we are using the following naming convention in ESA: `{prefix}-{environment}-{resource-type}{instance}`
For storage accounts, we are also including the layer in the name, i.e. `workspace`, `raw`, `enriched & curated`.

If users want to specify their own naming convention, then they are free to do so. If this is something that needs to be done, I would suggest to follow the other deployment option described [here](https://github.com/Azure/data-management-zone/blob/main/docs/EnterpriseScaleAnalytics-CreateRepository.md) and make the necessary changes to the bicep templates as part of the parameter update process.

**Comment by baatch** [comment]
Thanks @marvinbuss for the reply and I understand the aspect of giving each landing zone the freedom to decide their own naming convention and that tagging should be sufficient

I disagree and think that most projects want to follow your recommendations and that it is somewhat contradicting that the Enterprise-Scale solution doesn't follow the CAF suggested naming conventions. Could you explain the reason why the Enterprise-Scale solutions choose to not follow the recommended naming conventions and what naming conventions you are using instead?

**Reply by marvinbuss** [reply]
Today we are using the following naming convention in ESA: `{prefix}-{environment}-{resource-type}{instance}`
For storage accounts, we are also including the layer in the name, i.e. `workspace`, `raw`, `enriched & curated`.

One of the reasons why we are not using the convention in the link above is that all the details would drastically reduce the allowed size of the prefix. This would make it more difficult for users to find a prefix that is not already taken and therefore may hinder some of the deployments. This would then also hinder adoption of ESA. In addition, we believe that some of the proposed sections are not required. For example, the region of services and artifacts is put front and center even the default table view in the Azure Portal.

Lastly, I just want to emphasize that naming convention is not a critical design area. If any additional context is required for services, customers should use tags and enforce tags via policy (i.e. for cost management reasons, to specify the environment, etc.).

**Reply by marvinbuss** [reply]
@baatch I also added these details to my comment above. Please let us know, if more details are required. Happy to provide more details and thoughts. Otherwise, I would like to kindly ask you to mark the comment above as answer. Thank you!

**Reply by baatch** [reply]
@marvinbuss thanks and exactly the answers that I was looking for 😁
